### PR TITLE
[drivers]: [modem]: Fixes Dereference before null check #5…

### PR DESCRIPTION
**Pull request by jaiiarora**
Fixes Dereference before null check.

Fixes #529532

### CODE REVIEW

File: drivers/modem/gsm_ppp.c [hunk 1]
```diff
@@ -615,15 +615,16 @@ static struct net_if *ppp_net_if(void)
...
-	int ret;
 
 	if (ppp_dev == NULL) {
 		LOG_ERR("Cannot find PPP %s!", CONFIG_NET_PPP_DRV_NAME);
 		return;
 	}
 
+	const struct ppp_api *api;
+	struct net_if *iface = gsm->iface;
+	int ret;
```
Comment (rerickson1) [inline drivers/modem/gsm_ppp.c:626]: Variable definitions should always be at the top of the function.
Assigning values can be done later in the function.

**Review by rerickson1 — CHANGES_REQUESTED** [review]

**Comment by rerickson1** [comment]
Commits need to be squashed. Is there a useful change here? Per the PR there are no net changes now...

**Comment by kartben** [comment]
ping @jaiiarora

**Comment by kartben** [comment]
Hello @jaiiarora  -- could you please provide feedback here?

**Comment by kartben** [comment]
@jaiiarora , will you be able to look at this?

**Comment by jaiiarora** [comment]
Hi @kartben , Sorry for my late reply. Sure I will look into this

**Comment by jaiiarora** [comment]
Hi @kartben , Can you let me know the issue number. I aam not able to find it hence not able to make progress

**Comment by kartben** [comment]
It looks like you were trying to fix https://github.com/zephyrproject-rtos/zephyr/issues/59532, but this has been taken care of already

**Comment by kartben** [comment]
@jaiiarora Your PR was awkward as there were two commits cancelling one another, hence why it wasn't reviewed in time. I hope you will consider submitting other PRs in the future!

**Comment by jaiiarora** [comment]
Sure

**Pull request closed without merging**